src/sqlfluff/rules/L051.py

```diff
@@ -36,23 +43,52 @@ class Rule_L051(BaseRule):
         INNER JOIN baz;
     """
 
+    config_keywords = ["fully_qualify_join_types"]
+
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
-        """INNER JOIN must be fully qualified."""
+        """Fully qualify JOINs."""
+        # Config type hints
+        self.fully_qualify_join_types: str
+
         # We are only interested in JOIN clauses.
-        if context.segment.type != "join_clause":
+        if not context.segment.is_type("join_clause"):
             return None
 
-        # We identify non-lone JOIN by looking at first child segment.
-        if context.segment.segments[0].name != "join":
-            return None
+        join_clause_keywords = [
+            segment for segment in context.segment.segments if segment.type == "keyword"
+        ]
+
+        # We identify LEFT/RIGHT/OUTER JOIN and if the next keyword is JOIN.
+        if (
+            self.fully_qualify_join_types in ["outer", "both"]
+            and join_clause_keywords[0].name in ["right", "left", "full"]
+            and join_clause_keywords[1].name == "join"
```

Comment:
Is `name` always lowercase? This looks to be what previous version of the rule was using but curious if that was correct?

Reply:
Yes, `name` seem to always return lowercase (irrelevant of the code capitalisation). On the other hand `raw` returns the keyword with the capitalisation found in the SQL.

I must admit I didn't dig that far into the codebase yet and stuck to the original approach.